target/linux/ramips/patches-4.4/0701-mt7620-add-mib-stats.patch

```diff
@@ -0,0 +1,280 @@
++	MIB_DESC(1, MT7620_MIB_STATS_GDM2_TX_GBCNT, "GDM2_TX_GBCNT"),
++	MIB_DESC(1, MT7620_MIB_STATS_GDM2_TX_GPCNT, "GDM2_TX_GPCNT"),
++	MIB_DESC(1, MT7620_MIB_STATS_GDM2_TX_SKIPCNT, "GDM2_TX_SKIPCNT"),
++	MIB_DESC(1, MT7620_MIB_STATS_GDM2_TX_COLCNT, "GDM2_TX_COLCNT"),
++	MIB_DESC(1, MT7620_MIB_STATS_GDM2_RX_GBCNT, "GDM2_RX_GBCNT"),
++	MIB_DESC(1, MT7620_MIB_STATS_GDM2_RX_GPCNT, "GDM2_RX_GPCNT"),
++	MIB_DESC(1, MT7620_MIB_STATS_GDM2_RX_OERCNT, "GDM2_RX_OERCNT"),
++	MIB_DESC(1, MT7620_MIB_STATS_GDM2_RX_FERCNT, "GDM2_RX_FERCNT"),
++	MIB_DESC(1, MT7620_MIB_STATS_GDM2_RX_SERCNT, "GDM2_RX_SERCNT"),
++	MIB_DESC(1, MT7620_MIB_STATS_GDM2_RX_LERCNT, "GDM2_RX_LERCNT"),
++	MIB_DESC(1, MT7620_MIB_STATS_GDM2_RX_CERCNT, "GDM2_RX_CERCNT"),
++	MIB_DESC(1, MT7620_MIB_STATS_GDM2_RX_FCCNT, "GDM2_RX_FCCNT")
++};
++
++static const struct mt7xxx_mib_desc mt7620_port_mibs[] = {
++	MIB_DESC(1, MT7620_MIB_STATS_PORT_TGPCN,  "TxGPC"),
++	MIB_DESC(1, MT7620_MIB_STATS_PORT_TBOCN,  "TxBOC"),
++	MIB_DESC(1, MT7620_MIB_STATS_PORT_TGOCN,  "TxGOC"),
++	MIB_DESC(1, MT7620_MIB_STATS_PORT_TEPCN,  "TxEPC"),
++	MIB_DESC(1, MT7620_MIB_STATS_PORT_RGPCN,  "RxGPC"),
++	MIB_DESC(1, MT7620_MIB_STATS_PORT_RBOCN,  "RxBOC"),
++	MIB_DESC(1, MT7620_MIB_STATS_PORT_RGOCN,  "RxGOC"),
++	MIB_DESC(1, MT7620_MIB_STATS_PORT_REPC1N, "RxEPC1"),
++	MIB_DESC(1, MT7620_MIB_STATS_PORT_REPC2N, "RxEPC2")
++};
++
+ #define MT7621_MIB_COUNTER_BASE	0x4000
+ #define MT7621_MIB_COUNTER_PORT_OFFSET	0x100
+ #define MT7621_STATS_TDPC	0x00
+@@ -624,18 +714,6 @@ mt7530_get_port_link(struct switch_dev *
+ 	return 0;
+ }
+ 
+-static const struct switch_attr mt7530_global[] = {
+-	{
+-		.type = SWITCH_TYPE_INT,
+-		.name = "enable_vlan",
+-		.description = "VLAN mode (1:enabled)",
+-		.max = 1,
+-		.id = MT7530_ATTR_ENABLE_VLAN,
+-		.get = mt7530_get_vlan_enable,
+-		.set = mt7530_set_vlan_enable,
+-	},
+-};
+-
+ static u64 get_mib_counter(struct mt7530_priv *priv, int i, int port)
+ {
+ 	unsigned int port_base;
+@@ -683,6 +761,111 @@ static int mt7621_sw_get_port_mib(struct
+ 	return 0;
+ }
+ 
++static u64 get_mib_counter_7620(struct mt7530_priv *priv, int i)
++{
++	u64 t;
++
++	t = mt7530_r32(priv, MT7620_MIB_COUNTER_BASE + mt7620_mibs[i].offset);
++	if (mt7620_mibs[i].size == 2) {
++		u64 hi;
++
++		hi = mt7530_r32(priv, MT7620_MIB_COUNTER_BASE + mt7620_mibs[i].offset + 4);
++		t |= hi << 32;
++	}
++
++	return t;
++}
++
++static u64 get_mib_counter_port_7620(struct mt7530_priv *priv, int i, int port)
++{
++	unsigned int port_base;
++	u64 t;
++
++	port_base = MT7620_MIB_COUNTER_BASE_PORT +
++		    MT7620_MIB_COUNTER_PORT_OFFSET * port;
++
++	t = mt7530_r32(priv, port_base + mt7620_port_mibs[i].offset);
++	if (mt7620_mibs[i].size == 2) {
++		u64 hi;
++
++		hi = mt7530_r32(priv, port_base + mt7620_port_mibs[i].offset + 4);
++		t |= hi << 32;
++	}
++
++	return t;
++}
++
++static int mt7530_sw_get_mib(struct switch_dev *dev,
++				  const struct switch_attr *attr,
++				  struct switch_val *val)
++{
++	static char buf[4096];
```

Comment:
How could it be accepted to openwrt? That code was just adapted from neighbour method.